cli/src/main/java/com/devonfw/tools/ide/repo/CustomToolRepositoryImpl.java

```diff
@@ -162,39 +162,46 @@ public static CustomToolRepository of(IdeContext context) {
-        JsonArray jsonTools = requireArray(jsonRoot.get("tools"));
-        for (JsonValue jsonTool : jsonTools) {
-          JsonObject jsonToolObject = requireObject(jsonTool);
-          String name = getString(jsonToolObject, "name");
-          String version = getString(jsonToolObject, "version");
-          String url = getString(jsonToolObject, "url", defaultUrl);
-          boolean osAgnostic = getBoolean(jsonToolObject, "os-agnostic", Boolean.FALSE);
-          boolean archAgnostic = getBoolean(jsonToolObject, "arch-agnostic", Boolean.TRUE);
-          if (url.isEmpty()) {
-            throw new IllegalStateException("Missing 'url' property for tool '" + name + "'!");
-          }
-          // TODO
-          String checksum = null;
-          CustomTool customTool = new CustomTool(name, VersionIdentifier.of(version), osAgnostic, archAgnostic, url,
-              checksum, context.getSystemInfo());
-          tools.add(customTool);
+      readCustomToolsFromJson(context, customToolsJson);
+    }
+    return new CustomToolRepositoryImpl(context, tools);
+  }
+
+  public static CustomToolsJson readCustomToolsFromJson(IdeContext context, Path customToolsJsonPath) {
+    try (InputStream in = Files.newInputStream(customToolsJsonPath);
+        Reader reader = new InputStreamReader(in, StandardCharsets.UTF_8)) {
+
+      JsonReader jsonReader = Json.createReader(new BufferedReader(reader));
+      JsonStructure json = jsonReader.read();
+      JsonObject jsonRoot = requireObject(json);
+      String defaultUrl = getString(jsonRoot, "url", "");
+      JsonArray jsonTools = requireArray(jsonRoot.get("tools"));
+      List<CustomTool> customTools = new ArrayList<>();
+
+      for (JsonValue jsonTool : jsonTools) {
+        JsonObject jsonToolObject = requireObject(jsonTool);
+        String name = getString(jsonToolObject, "name");
+        String version = getString(jsonToolObject, "version");
+        String url = getString(jsonToolObject, "url", defaultUrl);
+        boolean osAgnostic = getBoolean(jsonToolObject, "os-agnostic", Boolean.FALSE);
+        boolean archAgnostic = getBoolean(jsonToolObject, "arch-agnostic", Boolean.TRUE);
+        if (url.isEmpty()) {
+          throw new IllegalStateException("Missing 'url' property for tool '" + name + "'!");
         }
-      } catch (Exception e) {
-        throw new IllegalStateException("Failed to read JSON from " + customToolsJson, e);
+        // TODO
+        String checksum = null;
+        CustomTool customTool = new CustomTool(name, VersionIdentifier.of(version), osAgnostic, archAgnostic, url,
+            checksum, context.getSystemInfo());
+        customTools.add(customTool);
       }
+      return new CustomToolsJson("tools", defaultUrl, customTools);
+    } catch (Exception e) {
+      throw new IllegalStateException("Failed to read JSON from " + customToolsJsonPath, e);
     }
-    return new CustomToolRepositoryImpl(context, tools);
   }
```

Comment:
This method and logic belong to `CustomToolsJson` class. Furhter, we can use Jackson for the mapping instead of writing the mapping code manually. We can add an internal method for post-processing that will set the `url` in all `CustomTool` instances to the parent one from `CustomToolsJson` in case it is `null`. This method can then be called both from the JSON parsing as well as from the String parsing method.